lib/shared/errors.js

```diff
@@ -0,0 +1,366 @@
+
+  },
+
+  /**
+   * @namespace EACCES
+   * @memberof HUMAN_FRIENDLY
+   */
+  EACCES: {
+
+    /**
+     * @property {Function} title
+     * @memberof EACCES
+     * @param {Error} error - error object
+     * @returns {String} message
+     */
+    title: _.constant('You don\'t have access to this resource'),
+
+    /**
+     * @property {Function} description
+     * @memberof EACCES
+     * @param {Error} error - error object
+     * @returns {String} message
+     */
+    description: _.constant('Please ensure you have to necessary permissions for this task')
+
+  },
+
+  /**
+   * @namespace ENOMEM
+   * @memberof HUMAN_FRIENDLY
+   */
+  ENOMEM: {
+
+    /**
+     * @property {Function} title
+     * @memberof ENOMEM
+     * @param {Error} error - error object
+     * @returns {String} message
+     */
+    title: _.constant('Your system ran out of memory'),
+
+    /**
+     * @property {Function} description
+     * @memberof ENOMEM
+     * @param {Error} error - error object
+     * @returns {String} message
+     */
+    description: _.constant('Make sure your system has enough available memory for this task')
+
+  }
+
+  /* eslint-enable new-cap */
+
+};
+
+/**
+ * @summary Get user friendly property from an error
+ * @function
+ * @private
+ *
+ * @param {Error} error - error
+ * @param {String} property - HUMAN_FRIENDLY property
+ * @returns {(String|Undefined)} user friendly message
+ *
+ * @example
+ * const error = new Error('My error');
+ * error.code = 'ENOMEM';
+ *
+ * const friendlyDescription = getUserFriendlyMessageProperty(error, 'description');
+ *
+ * if (friendlyDescription) {
+ *   console.log(friendlyDescription);
+ * }
+ */
+const getUserFriendlyMessageProperty = (error, property) => {
+  const code = _.get(error, 'code');
+
+  if (_.isNil(code) || !_.isString(code)) {
+    return;
+  }
+
+  return _.invoke(exports.HUMAN_FRIENDLY, [ code, property ], error);
+};
+
+/**
+ * @summary Check whether an error should be reported to TrackJS
+ * @function
+ * @public
+ *
+ * @description
+ * In order to determine whether the error should be reported, we
+ * check a property called `report`. For backwards compatibility, and
+ * to properly handle errors that we don't control, an error without
+ * this property is reported automatically.
+ *
+ * @param {Error} error - error
+ * @returns {Boolean} whether the error should be reported
+ *
+ * @example
+ * if (errors.shouldReport(new Error('foo'))) {
+ *   console.log('We should report this error');
+ * }
+ */
+exports.shouldReport = (error) => {
+  return !_.has(error, 'report') || Boolean(error.report);
+};
+
+/**
+ * @summary Check if a string is blank
+ * @function
+ * @private
+ *
+ * @param {String} string - string
+ * @returns {Boolean} whether the string is blank
+ *
+ * @example
+ * if (isBlank('   ')) {
+ *   console.log('The string is blank');
+ * }
+ */
+const isBlank = _.flow([ _.trim, _.isEmpty ]);
```

Comment:
The thing is that this code will run on error objects that we don't control, and they could really be anything, so I guess it's better to be defensive.